packages/frontend/core/src/modules/cmdk/entities/quick-search-manager.ts

```diff
@@ -0,0 +1,192 @@
+import type {
+  DocRecord,
+  DocsService,
+  WorkspaceLocalState,
+  WorkspaceService,
+} from '@toeverything/infra';
+import { Entity, LiveData } from '@toeverything/infra';
+import type { Observable } from 'rxjs';
+import { race, skip, Subject } from 'rxjs';
+
+import { resolveLinkToDoc } from '../../navigation';
+
+type QuickSearchMode = 'cmdk' | 'search-callback';
+
+const RECENT_PAGES_LIMIT = 3; // adjust this?
+const RECENT_PAGES_KEY = 'recent-pages';
+
+const EMPTY_ARRAY: string[] = [];
+
+export type SearchCallbackResult =
+  | {
+      docId: string;
+      blockId?: string;
+    }
+  | {
+      query: string;
+      action: 'insert';
+    };
+
+// todo: move command registry to entity as well
+export class QuickSearchManager extends Entity {
```

Comment:
How about name it `QuickSearch` or `QuickSearchPanel`, which can represent an "object" abstracted in our system.
Avoid names like Manager or Worker, as they sound more like a bunch of logic.
This is what I learned in domain-driven design, and I hope to follow such rules.